### PR TITLE
DataCollector Usage Experience

### PR DESCRIPTION
**Pull request by harshjain2**
DataCollector Usage Experience

### CODE REVIEW

File: docs/extensions/datacollector-usage.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Using DataCollectors
```
Comment (codito) [inline docs/extensions/datacollector-usage.md]: Please merge this doc with `analyze.md`, the earlier doc specifically is about Data Collectors.

File: docs/analyze.md [hunk 2]
```diff
@@ -1,124 +1,141 @@
...
+1. A brief overview of DataCollectors.
+2. Configuring DataCollectors in TPv2.
+3. Key differences for using DataCollectors in TPv2 v/s TPv1.
+3. Instructions for using [code coverage][coverage].
 
 > **Version note:**
 >
-> Data collectors are supported on test platform `15.3.0` onwards. It is part of
+> DataCollectors are supported on test platform `15.3.0` onwards. It is part of
 > VS 2017 15.3 and dotnet-cli 2.0.0 builds.
```
Comment (sbaid) [inline docs/analyze.md]: Pl add an appendix section clarifying >= build numbers where it is available and how to get that info from help->about in VS and dotnet --help etc.

File: docs/analyze.md [hunk 3]
```diff
@@ -1,124 +1,141 @@
...
-1. the `Extensions` folder along side `vstest.console.exe`. E.g. in case of 
-dotnet-cli, the path could be `/sdk/<version>/Extensions` directory.
-2. any well known location on the filesystem
- 
-> Version Note: TBD, draft spec
-In case of #2, user can specify the full path to the location using `/extensions:<path>`
-command line switch. Test platform will locate extensions from the provided
-directory.
  
 #### Naming
```
Comment (sbaid) [inline docs/analyze.md]: Delete the naming section also. it belongs to design/architecture document

File: docs/analyze.md [hunk 4]
```diff
@@ -1,124 +1,141 @@
...
+Below is the sample command to configure and use DataCollectors through vstest.console command line.
+```
+> vstest.console.exe test_project.dll /collect:"Code Coverage"
+> vstest.console.exe test_project.dll /collect:"MyDataCollector" /testadapterpath:<Path to MyDataCollector assembly>
+```
+Please note that `testadapterpath` is not required for DataCollectors shipped along with TPv2.
+
+### Using TestSettings
+While the recommended way is to use [runsettings](#Using-RunSettings) or [vstest.console args](#Using-vstest.console-args), there are few DataCollectors which only worked with testsettings.
+E.g.: `System Information` DataCollector. Below is the sample testsettings for using `System Information` DataCollctor.
```
Comment (sbaid) [inline docs/analyze.md]: typo DataCollctor

File: docs/analyze.md [hunk 5]
```diff
@@ -1,124 +1,141 @@
...
+
+```xml
```
Comment (sbaid) [inline docs/analyze.md]: pl try that the cut and paste of these settings work

File: docs/analyze.md [hunk 6]
```diff
@@ -1,124 +1,141 @@
...
+> vstest.console.exe test_project.dll /settings:datacollection.testsettings
+```
+### Enable/Disable a DataCollector
+All DataCollectors configured in the .runsettings files are loaded automatically and are enabled to participate for run, unless explicitely disabled using boolean valued attribute named `enabled`.
+
+For example, only `coverage` DataCollector will be enabled for a test run with
 below runsettings:
 
 ```xml
 <DataCollectionRunSettings>
```
Comment (sbaid) [inline docs/analyze.md]: pl remove code coverage partial section of runsettings. it has incorrect friendly name. also people may make mistake to add this section only as runsettings.

File: docs/analyze.md [hunk 7]
```diff
@@ -1,124 +1,141 @@
...
-A specific data collector can be explicitely enabled using the
-`/collect:<friendly name>` command line switch.
+A specific DataCollector can be explicitely enabled using the `/collect:<friendly name>` command line switch.
 
-For example, below command line will enable a data collector named `coverage` 
-(and disable other data collectors mentioned in .runsettings):
+For example, below command line will enable a DataCollector named `coverage` 
+(and disable other DataCollectors mentioned in .runsettings):
 ```
 > vstest.console test_project.dll /collect:coverage
```
Comment (sbaid) [inline docs/analyze.md]: incorrect friendly name for coverage

File: docs/analyze.md [hunk 8]
```diff
@@ -1,124 +1,141 @@
...
 ```
  
-More than one data collectors can also be enabled using `/collect` command line switch
+More than one DataCollectors can also be enabled using `/collect` command line switch
 
-For example, below command will enable data collectors named `coverage` and `systeminfo`:
+For example, below command will enable DataCollectors named `coverage` and `systeminfo`:
 ```
 > vstest.console test_project.dll /collect:coverage /collect:systeminfo
```
Comment (sbaid) [inline docs/analyze.md]: incorrect coverage friendly name

File: docs/analyze.md (hunk 4 above)
Comment (harshjain2) [inline docs/analyze.md]: Done.

File: docs/analyze.md (hunk 5 above)
Comment (harshjain2) [inline docs/analyze.md]: Cut and paste of these settings works as expected.

File: docs/analyze.md (hunk 6 above)
Comment (harshjain2) [inline docs/analyze.md]: Added complete runsettings.

Comment (harshjain2) [inline docs/analyze.md]: And corrected Friendly Name of Code Coverage

File: docs/analyze.md (hunk 7 above)
Comment (harshjain2) [inline docs/analyze.md]: Changed to `MyDataCollector1`

File: docs/analyze.md (hunk 3 above)
Comment (harshjain2) [inline docs/analyze.md]: Moved to datacollector.md

File: docs/analyze.md (hunk 8 above)
Comment (harshjain2) [inline docs/analyze.md]: Done.

**Pull request merged**